### PR TITLE
Shamir's Secret Sharing (SSS) strategy, filechecker restore command

### PR DESCRIPTION
**Pull request by papum20**
Implementation of two ranflood strategies, based on Shamir's Secret Sharing and only differing in configuration parameters, and of a subcommand for the filechecker tool for the subsequent restoration, after an attack.

### CODE REVIEW

**Review by thesave — COMMENTED** [review]
Thank you very much for your PR and implementation of the SSS flooders (and the additional fixes and features you added)!
I commented on a few files.
Everything else seems fine.
Before proceeding with the merge, I kindly ask you to add on top of the files you created the copyright and your reference, following the template, say, of the [SSSFloodTask.java](https://github.com/Flooding-against-Ransomware/ranflood/pull/5/files#diff-4cebdc4de53557b6715d36056a826e8bbdb3a4e7caa3780f67170a0ca32ad270) file.

File: build.gradle [hunk 1]
```diff
@@ -173,6 +173,48 @@ task filecheckerJar ( type: Jar ) {
     }
 }
 
+task testCompareJar ( type: Jar ) {
```
[the remark this answers is not in the document]
Comment (papum20) [reply build.gradle:176]: no they aren't: I left the tests i performed, but I can take care of removing all of them.

**Comment by papum20** [comment]
Ok, thank you. For the copyright headers to add to my files, should I replace `Copyright 2021 (C) by Saverio Giallorenzo <saverio.giallorenzo@gmail.com> ` with my reference (or just add my reference after)?

**Comment by papum20** [comment]
Ok, I removed test files, added copyrights with my reference and marked version 0.6.

**Comment by thesave** [comment]
Perfect, merged.
Thanks a lot!

**Pull request merged**